Change global search block to custom one

jabrah opened the pull request:
Closes https://github.com/jhu-idc/iDC-general/issues/111

Note that we've proposed removing the "active filters" section in the mockup linked in this issue and integrating the same functionality in the facets components in the left sidebar

## TODO 

* [x] Update commit hashes for UI theme and module when appropriate PRs are merged
  * Theme: https://github.com/jhu-idc/idc-ui-theme/pull/38
  * Module: https://github.com/jhu-idc/idc_ui_module/pull/11
* [x] Update snapshot to incorporate block change config 

## Changes

* Add new general search page at route `/search` -- replacing the previously used `/solr-search` route
* Replace simple global search with a custom one that goes to the correct route
* Use the same search component that is used on the collections list and collection details pages, bringing search query (with Solr query support), search options, facets, etc
* IDC search components can now be initialized with URL query parameters
  * `?query=` - init search term, appears in the input textbox
  * `?sort_order=` - init sort order in search options. Valid values: `ASC|DSC` for ascending or descending order
  * `?sort_by=` - init sortBy parameter, valid: `title|search_api_relevance`
  * `?page=` - init search results page
  * `?items_per_page=` - init items per page returned in the search results
  * `f[i]=` - add one or more facets, `i` is an integer, syntax similar to an array
    * Facet formatted as `facet_field:facet_value`
    * Example facet query params: `f[0]=subject:Moo`

## Testing

You'll need to add some data in order for search to do anything useful. At least one collection and at least two repo items (preferably each a member of the added collection) should be added to the system.

* You can enter a phrase into the global search component and click the search button.
  * You should end up at the `/search` route with your search phrase appended as a URL query parameter
  * The page will search according to your original query
* Changing search options or selecting facets should refresh the displayed results and update URL query parameters in your browser

## Preview

![image](https://user-images.githubusercontent.com/5167587/113613006-95b84980-961e-11eb-9115-b812b244a92b.png)

bseeger commented:
This looks good to me. The only issue I found was that if a user is on the search page and they want to start a new search, they have to clear the facets first, else they become part of the new search's URL.  Might be handy if we had a button that resets the search page.

jabrah commented:
@bseeger 

Added a "Clear" button separately for facets and list options. Example:  [![Image from Gyazo](https://i.gyazo.com/3ec806312e79324716fe06da4dcdafd1.gif)](https://gyazo.com/3ec806312e79324716fe06da4dcdafd1)

Also added a number of misc fixes:
* Address inconsistent widths for various components
* Add hover style to clickables on the home page
* Left align facet labels - they're current center aligned, which looks weird for certain screen sizes
* Modify labels/alt text for accessibility

bseeger reviewed (approved):
The changes look good to me.
